app/api/promiseMiddleware.js

```diff
@@ -0,0 +1,22 @@
+export default function promiseMiddleware() {
+  return next => action => {
+    const { promise, type, ...rest } = action;
+
+    if (!promise) return next(action);
+
+    const SUCCESS = type + '_SUCCESS';
+    const REQUEST = type + '_REQUEST';
+    const FAILURE = type + '_FAILURE';
+    next({ ...rest, type: REQUEST });
+    return promise
+      .then(req => {
+        next({ ...rest, req, type: SUCCESS });
+        return true;
+      })
+      .catch(error => {
+        next({ ...rest, error, type: FAILURE });
+        console.log(error);
+        return false;
+      });
+   };
+}
```

Comment:
Is it worth moving this to a folder called `/middlewares` in case we wish to add more middlewares in the future?